tests/data/virtual-server-canned-responses/standard/virtual-server.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: k8s.nginx.org/v1
+kind: VirtualServer
+metadata:
+  name: virtual-server
+spec:
+  host: virtual-server.example.com
+  routes:
+  - path: "/canned-response"
+    action:
+      return:
+        code: 407 # not required. the default is 200
+        type: application/json # not required. default is text/plain
+        body: '{
+                 \"host\": \"${host}\",
+                 \"request_time\": \"${request_time}\",
+                 \"pid\": \"${pid}\",
+                 \"server_protocol\": \"${server_protocol}\",
+                 \"connections_active\": \"${connections_active}\",
+                 \"connections_writing\": \"${connections_writing}\",
+                 \"request_uri\": \"${request_uri}\",
+                 \"remote_addr\": \"${remote_addr}\",
+                 \"remote_port\": \"${remote_port}\",
+                 \"server_addr\": \"${server_addr}\",
+                 \"request_method\": \"${request_method}\",
+                 \"scheme\": \"${scheme}\",
+                 \"request_length\": \"${request_length}\",
+                 \"nginx_version\": \"${nginx_version}\",
+                 \"connection\": \"${connection}\",
+                 \"time_local\": \"${time_local}\",
+                 \"server_port\": \"${server_port}\",
+                 \"server_name\": \"${server_name}\",
+                 \"connections_waiting\": \"${connections_waiting}\",
+                 \"request_body\": \"${request_body}\",
+                 \"args\": \"${args}\",
+                 \"time_iso8601\": \"${time_iso8601}\",
+                 \"connections_reading\": \"${connections_reading}\"
+               }'
+  - path: "/canned-defaults"
+    action:
+      return:
+        code:
+        type:
+        body: >
```

Comment:
this transforms into a string with one line "line1 line2 line3". if you want to test for multi-line string, I suggest changing to:
```yaml
body: |
  line 1
  line 2
  line 3
```